components/table/table-wrapper.js

```diff
@@ -359,19 +361,54 @@ export class TableWrapper extends RtlMixin(SelectionMixin(LitElement)) {
 				this._tableIntersectionObserver = new IntersectionObserver((entries) => {
 					entries.forEach((entry) => {
 						if (entry.isIntersecting) {
-							this._applyClassNames(table);
+							this._handleTableChange();
 						}
 					});
 				});
 			} else {
 				this._tableIntersectionObserver.disconnect();
 			}
-			this._tableIntersectionObserver.observe(table);
+			this._tableIntersectionObserver.observe(this._table);
 		}
 
-		this._applyClassNames(table);
+		this._handleTableChange();
+	}
+
+	async _handleTableChange() {
+		await new Promise(resolve => requestAnimationFrame(resolve));
+
+		this._applyClassNames();
+		this._updateStickyTops();
+	}
 
+	_registerMutationObserver(observerName, callback, target, options) {
+		if (this[observerName]) {
+			this[observerName].disconnect();
+		} else if (target) {
+			this[observerName] = new MutationObserver(callback);
+		}
+		if (target) this[observerName].observe(target, options);
 	}
+
+	_updateStickyTops() {
+		if (!this._table || !this.stickyHeaders) return;
+
+		const controlsHeight = this._controls && !this._controls.noSticky ? this._controls.offsetHeight + 4 : 0; // +4 for the shadow's `bottom: -4px`.
+		const borderRadius = this.type === 'default' ? 5 : 0;
```

Comment:
Wonder if it's worth defining this as a `const` above and then using it here and in the CSS? Weirdly it looks like there's a mix of `5px` and `0.3rem` (`6px` @ `20px` font size)...

Reply:
Haha true, I'll investigate that inconsistency...

Reply:
If we need to switch that from `rem`-based to `px`-based in the variable definition, that's likely fine. We've been doing a bit of that for other components lately for the WCAG 2.2 work.